### PR TITLE
feat(#11752): The contentPath of AddressServerUrl is not flexible.

### PR DESCRIPTION
**Pull request by Bo-Qiu**
## What is the purpose of the change

feat(#11752 ): The contentPath of AddressServerUrl is not flexible.

## Brief changelog

feat(#11752 ): The contentPath of AddressServerUrl is not flexible.

## Verifying this change

feat(#11752 ): The contentPath of AddressServerUrl is not flexible.

### CODE REVIEW

File: client/src/main/java/com/alibaba/nacos/client/config/impl/ServerListManager.java [hunk 1]
```diff
@@ -161,34 +156,24 @@ public ServerListManager(String endpoint) throws NacosException {
...
         this.isStarted = false;
-        Properties properties = new Properties();
-        properties.setProperty(PropertyKeyConst.ENDPOINT, endpoint);
-        final NacosClientProperties clientProperties = NacosClientProperties.PROTOTYPE.derive(properties);
-        this.endpoint = initEndpoint(clientProperties);
-        
         if (StringUtils.isBlank(endpoint)) {
             throw new NacosException(NacosException.CLIENT_INVALID_PARAM, "endpoint is blank");
         }
+        Properties properties = new Properties();
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/config/impl/ServerListManager.java:162]: 这个移下来不合理， endpoint有可能并非通过参数直接传入，可能通过环境变量等传入。需要保留原来的逻辑。

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: client/src/main/java/com/alibaba/nacos/client/config/impl/ServerListManager.java (hunk 1 above)
Comment (Bo-Qiu) [reply client/src/main/java/com/alibaba/nacos/client/config/impl/ServerListManager.java:162]: 原代码是判断的方法入参endpoint是否为空，所以调整顺序没影响。 我理解你是期望这儿判断this.endpoint是否为空吗？
image

Comment (KomachiSion) [reply client/src/main/java/com/alibaba/nacos/client/config/impl/ServerListManager.java:162]: OK

**Review by KomachiSion — CHANGES_REQUESTED** [review]
How about the ServerListManager in client/naming/package?

**Comment by Bo-Qiu** [comment]
> How about the ServerListManager in client/naming/package?

I changed it, please review it again, thx!

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**